main.tf

```
@@ -1,13 +1,203 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+provider "awsutils" {
+  region = var.region
+}
+
+locals {
+    enabled = module.this.enabled
+    mutual_enabled = 
+}
+
+module "self_signed_cert_ca" {
+  source = "cloudposse/ssm-tls-self-signed-cert/aws"
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+  name = "self-signed-cert-ca"
+
+  subject = {
+    common_name  = module.this.id
+    organization = var.organization_name
+  }
+
+  basic_constraints = {
+    ca = true
   }
+
+  allowed_uses = [
+    "crl_signing",
+    "cert_signing",
+  ]
+
+  context = module.this.context
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+module "self_signed_cert_root" {
+  source = "cloudposse/ssm-tls-self-signed-cert/aws"
+
+  name = "self-signed-cert-root"
+
+  subject = {
+    common_name  = module.this.id
+    organization = var.organization_name
+  }
+
+  basic_constraints = {
+    ca = false
+  }
+
+  allowed_uses = [
+    "key_encipherment",
+    "digital_signature",
+    "server_auth"    
+  ]
+
+  context = module.this.context
+}
+
+module "self_signed_cert_server" {
+  source = "cloudposse/ssm-tls-self-signed-cert/aws"
+
+  name = "self-signed-cert-server"
+
+  subject = {
+    common_name  = module.this.id
+    organization = var.organization_name
+  }
+
+  basic_constraints = {
+    ca = false
+  }
+
+  allowed_uses = [
+    "key_encipherment",
+    "digital_signature",
+    "server_auth"    
+  ]
+
+  context = module.this.context
+}
+
+resource "aws_acm_certificate" "ca" {
+  private_key      = module.self_signed_cert_ca.certificate_key_path
+  certificate_body = module.self_signed_cert_ca.certificate_pem
+}
+
+resource "aws_acm_certificate" "root" {
+  private_key       = module.self_signed_cert_root.certificate_key_path
+  certificate_body  = module.self_signed_cert_root.certificate_pem
+  certificate_chain = module.self_signed_cert_ca.certificate_pem
+}
+
+resource "aws_acm_certificate" "server" {
+  private_key       = module.self_signed_cert_server.certificate_key_path
+  certificate_body  = module.self_signed_cert_server.certificate_pem
+  certificate_chain = module.self_signed_cert_ca.certificate_pem
+}
+
+module "cloudwatch_log" {
+  source = "cloudposse/cloudwatch-logs/aws"
+
+  stream_names = [ var.logging_stream_name ]
+
+  context = module.this.context
+}
+
+resource "aws_ec2_client_vpn_endpoint" "default" {
+  description            = module.this.id
+  server_certificate_arn = aws_acm_certificate.server.arn
+  client_cidr_block      = var.client_cidr
+
+  authentication_options = local.mutual_enabled ? {
+    type                       = "certificate-authentication"
+    root_certificate_chain_arn = aws_acm_certificate.root.arn
+  } : {
+    type              = "federated-authentication"
+    saml_provider_arn = try(aws_iam_saml_provider.this[0].arn, var.saml_provider_arn)
+  }
```

do something like....

```suggestion
  authentication_options = local.authentication_type == "certificate-authentication" ? {
    type                       = "certificate-authentication"
    root_certificate_chain_arn = aws_acm_certificate.root.arn
  } : {
    type              = "federated-authentication"
    saml_provider_arn = try(aws_iam_saml_provider.this[0].arn, var.saml_provider_arn)
  }
```